test/functional/pdo_sqlsrv/pdo_ae_insert_pdoparam_numeric.phpt

```diff
@@ -1,39 +1,123 @@
-            }
-            if ($r === false) {
-                isIncompatibleTypesError($stmt, $dataType, $pdoParamType);
+            $stmt = insertRow($conn, $tbname, array( "c_det" => new BindParamOp(1, $inputValues[0], $pdoParamType), "c_rand" => new BindParamOp(2, $inputValues[1], $pdoParamType)), "prepareBindParam", $r);
+            
+            // check the case when inserting as PDO::PARAM_NULL
+            // with or without AE: NULL is inserted
+            if ($pdoParamType == "PDO::PARAM_NULL") {
+                if ($r === false) {
+                    echo "Conversion from $pdoParamType to $dataType should be supported\n";
+                } else {
+                    $sql = "SELECT c_det, c_rand FROM $tbname";
+                    $stmt = $conn->query($sql);
+                    $row = $stmt->fetch(PDO::FETCH_ASSOC);
+                    if (!is_null($row['c_det']) || !is_null($row['c_rand'])) {
+                        echo "Conversion from $pdoParamType to $dataType should insert NULL\n";
+                    }
+                }
+            // check the case when inserting as PDO::PARAM_BOOL
+            // with or without AE: 1 or 0 should be inserted when inserting into an integer column
+            //                     double is inserted when inserting into a real column
+            } else if ($pdoParamType == "PDO::PARAM_BOOL") {
+                if ($r === false) {
+                    echo "Conversion from $pdoParamType to $dataType should be supported\n";
+                } else {
+                    $sql = "SELECT c_det, c_rand FROM $tbname";
+                    $stmt = $conn->query($sql);
+                    $row = $stmt->fetch(PDO::FETCH_ASSOC);
+                    if ($dataType == "real") {
+                        if (abs($row['c_det'] - $inputValues[0]) < $epsilon && abs($row['c_rand'] - $inputValues[1]) < $epsilon) {
+                            echo "****Conversion from $pdoParamType to $dataType is supported****\n";
+                        } else {
+                            echo "Conversion from $pdoParamType to $dataType causes data corruption\n";
+                        }
+                    } else {
+                        if ($row['c_det'] != ($inputValues[0] != 0) && $row['c_rand'] != ($inputValues[1] != 0)) {
+                            echo "Conversion from $pdoParamType to $dataType insert a boolean\n";
+                        }
+                    }
+                }
+            // check the case when inserting as PDO::PARAM_STR into a bigint column
+            // with AE: should work
+            // without AE: should not work on a x86 platform
+            } else if ($dataType == "bigint" && $pdoParamType == "PDO::PARAM_STR") {
+                if (!isAEConnected() && PHP_INT_SIZE == 4) {
+                    if ($r !== false) {
+                        echo "Conversion from $pdoParamType to $dataType should not be supported\n";
```

Comment:
Consider changing the error message to "Conversion from .... should have failed."

Reply:
I want to be consistent with the other comments.